policies/legacy-migration.md

```diff
@@ -0,0 +1,42 @@
+# Legacy Provider Policy
+
+## Purpose
+The Legacy Provider exists to create an opt-in availability mechanism for
+algorithms that, for various reasons, should have their use discouraged.  These
+reasons include, but are not limited to:
+
+* Discovered security issues leaving the algorithm in question unsafe for
+  general use
+
+* Lack of popular use (i.e. balancing code size vs consumption frequency)
+
+OpenSSL recognizes that consumption of these algorithms may continue to be
+required by consuming applications after the conditions above have been
+recognized.  The Legacy provider exists to provide a mechanism for such
+applications to continue having access to these algorithms while preventing
+applications that don't require them from inadvertently using them.
+
+## Constraints on moving an algorithm to the legacy provider
+
+1) Migration of an algorithm to the legacy provider must occur on a semantically
+versioned major release boundary.  Once a major release includes a given
+algorithm in a given provider, it must remain there for every minor release in
+that major stream.
+
+2) Prior to migration, the migration must be announced for at least 1
+semantically versioned minor release (see announcement mechanism below).
+
+3) Coincidental to the announcement above, the algorithm in question may be made
+available in both the source provider and the legacy provider.
```

Comment:
source provider?  Did you mean default provider?

Reply:
I used the term source, as I didn't want to constrain deprecation to only the default provider.  i.e. if we ever want to deprecate algorithms in the fips provider, or if we create say a pq provider in the future, this same process might be useable.

Reply:
This is the *only* place this file says "source provider".  Everywhere else, this file says "default provider"...  would that make you reconsider?

Reply:
It does make me reconsider, but I think its an open question of wording.  do we want to constrain this policy to the default provider only, or should it apply to any non-legacy provider (default/base/fips/etc)?

Reply:
Last time I checked, `default` has all the implementations that are in `base` and `fips` combined.  `base` is just there to be combined with `fips` to give it support stuff that can't be in the latter.

Reply:
Looking back at this, I would go for consistency.  Either "default provider", or "source provider", everywhere.